thirdeye-ui/src/app/pages/impact-dashboard-page/data/use-api-requests.tsx

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2024 StarTree Inc
+ *
+ * Licensed under the StarTree Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at http://www.startree.ai/legal/startree-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT * WARRANTIES OF ANY KIND,
+ * either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+import { useEffect, useState } from "react";
+
+// Interfaces
+import { APIRequestData, ApiRequestsProps } from "./api-requests.interfaces";
+
+// APIs
+import { useGetSubscriptionGroups } from "../../../rest/subscription-groups/subscription-groups.actions";
+import {
+    useGetAnomalies,
+    useGetAnomaly,
+} from "../../../rest/anomalies/anomaly.actions";
+import { useGetInvestigations } from "../../../rest/rca/rca.actions";
+
+// Utils
+import {
+    anaylysisPeriodPreviousWindowTimeMapping,
+    anaylysisPeriodStartTimeMapping,
+} from "../../../platform/utils";
+import { notifyIfErrors } from "../../../utils/notifications/notifications.util";
+import { useGetAlerts } from "../../../rest/alerts/alerts.actions";
+
+import { useNotificationProviderV1 } from "../../../platform/components";
+import { useTranslation } from "react-i18next";
+import { isEmpty } from "lodash";
+
+export const useApiRequests = ({
+    selectedAnalysisPeriod,
+}: ApiRequestsProps): APIRequestData => {
+    const { t } = useTranslation();
+    const { notify } = useNotificationProviderV1();
+    const [recentAnomalyInvestigatedId, setRecentAnomalyInvestigatedId] =
+        useState<number>();
+    const {
+        subscriptionGroups,
+        getSubscriptionGroups,
+        status: subscriptionGroupsStatus,
+        errorMessages: subscriptionGroupsErrorMessages,
+    } = useGetSubscriptionGroups();
+
+    const {
+        anomaly: mostRecentlyInvestigatedAnomaly,
+        getAnomaly,
+        status: anomalyStatus,
+        errorMessages: anomalyErrorMessages,
+    } = useGetAnomaly();
+
+    const {
+        alerts,
+        getAlerts,
+        status: alertsCountStatus,
+        errorMessages: alertsCountErrorMessages,
+    } = useGetAlerts();
+    const {
+        anomalies,
+        getAnomalies,
+        status: anomaliesStatus,
+        errorMessages: anomaliesErrorMessages,
+    } = useGetAnomalies();
+
+    const {
+        anomalies: previousPeriodAnomalies,
+        getAnomalies: getPreviousPeriodAnomalies,
+        status: previousPeriodAnomaliesStatus,
+        errorMessages: previousPeriodAnomaliesErrorMessages,
+    } = useGetAnomalies();
+
+    const {
+        investigations,
+        getInvestigations,
+        status: investigationStatus,
+        errorMessages: investigationErrorMessages,
+    } = useGetInvestigations();
+
+    useEffect(() => {
```

Comment:
Is there a better way to handle this? If you are using RQ, you can probably use the `onError` callback for handling errors and showing error alerts.